pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1091,8 +1091,8 @@ private boolean isPriorBatchIndex(long idx) {
         return resetIncludeHead ? idx < startMessageId.getBatchIndex() : idx <= startMessageId.getBatchIndex();
     }
 
-    private boolean isResetIncludedAndSameEntryLedger(MessageIdData messageId) {
-        return !resetIncludeHead && startMessageId != null
+    private boolean isNonDurableAndSameEntryAndLedger(MessageIdData messageId) {
```

Comment:
Due to this method does not only support for non-durable subscription, so is it better to named `isSameEntry()`?

Reply:
Fixed.